test/v2/services/ObjectStoreServiceSpec.scala

```diff
@@ -74,13 +78,39 @@ class ObjectStoreServiceSpec
       arbitraryMovementId.arbitrary,
       arbitraryMessageId.arbitrary
     ) {
-      (movemementId, messageId) =>
-        val result = objectStoreService.addMessage(DownloadUrl("invalidURL"), movemementId, messageId)
+      (movementId, messageId) =>
+        val result = objectStoreService.addMessage(DownloadUrl("invalidURL"), movementId, messageId)
 
         whenReady(result.value) {
           case Right(_) => fail("should have returned a Left")
           case Left(x)  => x
         }
     }
   }
+
+  "On getting a message from object store" - {
+    "given a successful response from the connector, should return a Right with Object Store Summary" in {
+      val movementId = MovementId("308c4a68e2cdc08f")
+      val messageId  = MessageId("123c4a68e2ele08f")
+      objectStoreService.addMessage(DownloadUrl("https://bucketName.s3.eu-west-2.amazonaws.com"), movementId, messageId)
```

Comment:
I think we can remove this part. not required as you are explicitly stubbed the getObject so you are always going to get something without adding anything in object store.

Reply:
> you are explicitly stubbed the getObject

Is that part of `ObjectStoreStub` -- does that always return something?

Reply:
Yes it is. if we are passing some uri that does not match path then it returns None.

Reply:
`ObjectStoreStub` isn't ours though.

If this works, then I'm happy to keep it as is.

Reply:
It is written by us if you can see that in fake folder and changed that with this PR only.

Reply:
No it doesn't, not the `ObjectStoreStub`, which is part of the client -- https://github.com/hmrc/object-store-client/blob/56d4e7e73aebeaa40c9284bc01bb65c7d34d1db7/object-store-client-play-28/src/main/scala/uk/gov/hmrc/objectstore/client/play/test/stub.scala#L78-L82.

You're thinking of the object-store SERVER stubbing, which uses the file system.

Reply:
Oh, I see what you were saying, I'd missed the part of your message saying _we'd_ written it, right.

I get what you're saying now -- https://github.com/hmrc/common-transit-convention-traders/pull/397/files/5647541c21e5a93b8ab7c2def660d355ae518062#diff-ed5f8f2141ae725f9526396be4fc46d06a963bcd0d8e7063b1ff807363f7414eR81.

Yup, I get it now -- yeah, putting the object can be ignored.

Reply:
Yes it's great that you were able to just use their stub @sonisn1988 and I massively envy that. I went thought the headache of trying to replicate what happens in the stub as much as possible for in our own stub that overrides theirs so that the functionality in the put would work as expected. https://github.com/hmrc/object-store-client/blob/56d4e7e73aebeaa40c9284bc01bb65c7d34d1db7/object-store-client-play-28/src/main/scala/uk/gov/hmrc/objectstore/client/play/test/stub.scala#L96

What happens is that they have an internal object-store map to store these values, while I see your solution to [hard code](https://github.com/hmrc/transit-movements-router/blob/main/test/uk/gov/hmrc/transitmovementsrouter/fakes/objectstore/ObjectStoreStub.scala#L64) it is one way to do it, I preferred writing the code with a map such that we can first of all replicate  what happens in the stub as much as possible but second of all have flexibility. I can then ensure that my put works as expected too.

@drnaylor-oc for your understanding

Reply:
Got it, thanks.